custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -58,206 +56,138 @@ def get_role_from_username(self, username):
-                                                                  'assigned_location_ids',
-                                                                  'last_login'])
-        return [u for u in user_query.run().hits]
-
-    def get_names_from_ids(self, location_id):
-        return location_id.split('_')[0] + '_name'
-
-    def convert_rs_to_matrix(self, results_queryset, main_location_type=None):
+    def populate_location_id_vs_name_mapping(self, results_queryset, main_location_type=None):
         """
-
         :param results_queryset: AwcLocation values queryset
         :param main_location_type: parent location type to fetch only locations below that level
-        :return: location matrix similar to a dataframe and locations ids of all the locations below the given
+        :return: dict with location id and value mapping and locations ids of all the locations below the given
          location
         """
-        location_matrix = []
         location_ids = []
-        if main_location_type is not None:
-            sub_location_types = self.location_types[self.location_types.index(main_location_type) + 1:]
-        else:
-            sub_location_types = self.location_types
-
         for result in results_queryset:
-            row_data = [result['state_id'], result['state_name'], result['district_id'],
-                        result['district_name'], result['block_id'], result['block_name']]
-            location_matrix.append(row_data)
-            # adding only descendants to the main location
-            for sub_location in sub_location_types:
+            for sub_location in self.location_types:
                 if result[sub_location] not in location_ids:
                     location_ids.append(result[sub_location])
-        return location_matrix, location_ids
+                    # adding location id and name to the dict
+                    self.location_id_name_dict[result[sub_location]] = result[
+                        self.get_location_type_string_from_location_id(sub_location) + '_name']
+        return location_ids
 
     def get_location_id_string_from_location_type(self, location_type):
+        """
+        Appends location _id the location
+        :param location_type:
+        :return:
+        """
         return location_type + '_id'
 
     def get_location_type_string_from_location_id(self, location_id):
         return location_id.replace('_id', '')
 
     def check_if_date_in_last_week(self, date):
+        """
+        :param date: user last activity date
+        :return: Returns if the user has an activity in dashboard in last 7 days
+        """
         if date is None:
             return 'N/A', False
-        date_formatted = dateutil.parser.parse(date).strftime("%d/%m/%Y, %I:%M %p")
-        d = datetime.datetime.strptime(date_formatted.split(',')[0], "%d/%m/%Y")
+        date_formatted = datetime.datetime.strftime(date, "%d/%m/%Y, %I:%M %p")
```

Comment:
Also this date returned from activity report is in IST you will have convert it to IST.